app/src/features/apiClient/screens/apiClient/components/views/http/components/HttpClientUrl/HttpClientUrl.tsx

```diff
@@ -15,15 +17,40 @@ interface ApiClientUrlProps {
   currentEnvironmentVariables: ScopedVariables;
   onEnterPress: (e: KeyboardEvent) => void;
   onUrlChange: (value: string, finalParams: KeyValuePair[]) => void;
+  onImportCurlRequest?: (request: RQAPI.Request) => void;
 }
 
-const HttpApiClientUrl = ({ url, currentEnvironmentVariables, onEnterPress, onUrlChange }: ApiClientUrlProps) => {
+//prettier-ignore
+const HttpApiClientUrl = ({ url, currentEnvironmentVariables, onEnterPress, onUrlChange, onImportCurlRequest }: ApiClientUrlProps) => {
   const [queryParams, setQueryParams] = useQueryParamStore((state) => [state.queryParams, state.setQueryParams]);
 
   const updatePathVariableKeys = usePathVariablesStore((state) => state.updateVariableKeys);
 
   const handleUrlChange = useCallback(
     (value: string) => {
+      const trimmedValue = value.trim();
```

Comment:
Triggering curl import on url change might lead to weird experience if user is in the middle of typing a curl by hand. 

So we should trigger the curl import logic only on paste